static/app/components/smartSearchBar/actions.tsx

```diff
@@ -0,0 +1,209 @@
+import {browserHistory, withRouter, WithRouterProps} from 'react-router';
+import styled from '@emotion/styled';
+
+import {openModal} from 'app/actionCreators/modal';
+import {pinSearch, unpinSearch} from 'app/actionCreators/savedSearches';
+import Access from 'app/components/acl/access';
+import Button from 'app/components/button';
+import MenuItem from 'app/components/menuItem';
+import {IconAdd, IconPin, IconSliders} from 'app/icons';
+import {t} from 'app/locale';
+import {SavedSearch} from 'app/types';
+import {trackAnalyticsEvent} from 'app/utils/analytics';
+import CreateSavedSearchModal from 'app/views/issueList/createSavedSearchModal';
+
+import SmartSearchBar from './index';
+import {removeSpace} from './utils';
+
+type SmartSearchBarProps = React.ComponentProps<typeof SmartSearchBar>;
+
+type ActionItem = NonNullable<SmartSearchBarProps['actionBarItems']>[number];
+type ActionProps = React.ComponentProps<ActionItem['Action']>;
+
+type PinSearchActionOpts = {
+  /**
+   * The currently pinned search
+   */
+  pinnedSearch?: SavedSearch;
+  /**
+   * The current issue sort
+   */
+  sort: string;
+};
+
+/**
+ * The Pin Search action toggles the current as a pinned search
+ */
+export function makePinSearchAction({pinnedSearch, sort}: PinSearchActionOpts) {
+  const PinSearchAction = ({
+    menuItemVariant,
+    savedSearchType,
+    organization,
+    api,
+    query,
+    location,
+  }: ActionProps & WithRouterProps) => {
+    const onTogglePinnedSearch = async (evt: React.MouseEvent) => {
+      evt.preventDefault();
+      evt.stopPropagation();
+
+      if (savedSearchType === undefined) {
+        return;
+      }
+
+      // eslint-disable-next-line no-unused-vars
+      const {cursor: _cursor, page: _page, ...currentQuery} = location.query;
+
+      trackAnalyticsEvent({
+        eventKey: 'search.pin',
+        eventName: 'Search: Pin',
+        organization_id: organization.id,
+        action: !!pinnedSearch ? 'unpin' : 'pin',
+        search_type: savedSearchType === 0 ? 'issues' : 'events',
```

Comment:
Hmm should we add a comment to L#50 lest someone tries to change it to `!savedSearchType`

Reply:
This should just use the enum

Reply:
Really we should have a map to keys or something, but 🤷